wiki/osu!_File_Formats/Osu_(file_format)/en.md

```diff
@@ -133,29 +133,76 @@ BeatmapSetID (Integer) is the ID of the beatmap set.
+The following 4 difficulty properties are expressed in *stars*, where 0 star means very easy, and 10 stars means very hard.
 
-`HPDrainRate:5`
+```
+HPDrainRate:4
+CircleSize:3.3
+OverallDifficulty:4
+ApproachRate:4.8
+```
+
+*HPDrainRate* specifies how fast the health bar decreases. At the time of this writing, the near future definition of this property remains unsettled.
+
+### Circle Size
+
+*CircleSize* (CS) defines the size of the hit circles and sliders in the osu!standard mode.
 
-CircleSize (Decimal) specifies the size of hit object circles.
+The resulting circle radius in osu!pixels is defined by the formula `32 * (1 - 0.7 * (CircleSize - 5) / 5)`, alternatively written `54.4 - 4.48 * CircleSize`.
 
-`CircleSize:4`
+Ranked beatmaps must have a CircleSize value between 2 and 7, inclusive.
 
-OverallDifficulty (Decimal) specifies the amount of time allowed to click a hit object on time.
+In osu!mania mode, *CircleSize* is the number of columns.
 
-`OverallDifficulty:6`
+### Overall Difficulty
 
-ApproachRate (Decimal) specifies the amount of time taken for the approach circle and hit object to appear.
+*OverallDifficulty* (OD) is the harshness of the hit window and the difficulty of spinners.
 
-`ApproachRate:7`
+- When OD < 5: `spins_per_second = 5 - 2 * (5 - OD) / 5`
+- When OD = 5: `spins_per_second = 5`
+- When OD > 5: `spins_per_second = 5 + 2.5 * (OD - 5) / 5`
 
-SliderMultiplier (Decimal) specifies a multiplier for the slider velocity. Default value is 1.4 .
+The width of the hit window is defined by the following table:
 
-`SliderMultiplier:1.3`
+| Score | Hit Window |
+| --- | --- |
+| 50 | `150ms + 50ms * (5 - OD) / 5` |
+| 100 | `100ms + 40ms * (5 - OD) / 5` |
+| 300 | `50ms + 30ms * (5 - OD) / 5` |
+
+### Approach Rate
+
+*ApproachRate* (AR) defines when hit objects start to fade in relative to when they should be hit.
+
+```
+                                        X = perfect hit
+                p r e e m p t           ↓
+ ├───────────────────────┬──────────────┤
+ 0%      fade_in           100% opacity
+```
+
+The circle starts fading in at `X - preempt` with:
+
+- When AR < 5: `preempt = 1200ms + 600ms * (5 - AR) / 5`
+- When AR = 5: `preempt = 1200ms`
+- When AR > 5: `preempt = 1200ms - 750ms * (AR - 5) / 5`
+
+The amount of time it takes for the hit object to completely fade in is also reliant on the approach rate:
+
+- When AR < 5: `fade_in = 800ms + 400ms * (5 - AR) / 5`
+- When AR = 5: `fade_in = 800ms`
+- When AR > 5: `fade_in = 800ms - 500ms * (AR - 5) / 5`
+
+### Sliders
+
+```
+SliderMultiplier:1.3
+SliderTickRate:1
+```
 
-SliderTickRate (Decimal) specifies how often slider ticks appear. Default value is 1.
+SliderMultiplier (Decimal) specifies a multiplier for the slider velocity. The reference velocity at slider multiplier = 1 is 100 osu!pixels per beat. A slider multiplier of 2 would yield a velocity of 200 osu!pixels per beat. The default slider multiplier is 1.4 when the property is omitted.
```

Comment:
specifies a multiplier for the slider velocity -> specifies the multiplier of the slider velocity
delete reference